### ISSUE TITLE
browserWindow has no icon

### ISSUE THREAD

**Issue by Kilian** [brave/browser-laptop#1282]
browserWindow needs an icon for brave to show an icon on Ubuntu or you get a grey icon:
![selection_001](https://cloud.githubusercontent.com/assets/41970/14382939/f1b98076-fd91-11e5-8a83-4ca94d8ebc51.png)

This should be added here: https://github.com/brave/browser-laptop/blob/75c788aeb00b82957b7be58473332df5da13ac80/js/stores/appStore.js#L111 and it should be a png.

**Comment by zvezdaburya** [comment]
Both the shortcut and browser while running has the icon showing for me. 
![braveiconshow](https://cloud.githubusercontent.com/assets/18287944/14388450/7a125184-fdae-11e5-96e1-611a19944645.png)

I am not sure what could be wrong.. is the shortcut for brave displaying the icon?

**Comment by luixxiul** [comment]
Confirmed, @bbondy Is branding for Linux ongoing?

**Comment by bbondy** [comment]
would love a PR to fix this @luixxiul

**Comment by luixxiul** [comment]
On Ubuntu it has been already implemented with a deb package.
![](https://cloud.githubusercontent.com/assets/3362943/15471805/df9491c8-2132-11e6-8bb3-bbf46cf6215e.jpg)

**Comment by bbondy** [comment]
I'm using 14.04 and it still comes up as unidentified for me.

**Comment by luixxiul** [comment]
Are you running brave with the deb package right? From console it comes up like that

**Comment by bbondy** [comment]
yep comes up without the icon for that. Maybe something is messed on my machine or maybe there's a bug.

**Comment by Kilian** [comment]
if you are not running from a .desktop file that supplies the icon, electron defaults back to the icon specified for the BrowserWindow. This is why in my original report I gave the line number of your BrowserWindow code. Simply add an icon property to it and link a png file, and that will fix this problem directly in your code instead of in a platform specific desktop file.

See also https://kilianvalkhof.com/talks/cross-platform-electron/#/11 :)

**Comment by luixxiul** [comment]
@Kilian, so is it OK with this?

https://github.com/brave/browser-laptop/blob/master/js/stores/appStore.js#L115

````
  let mainWindow = new BrowserWindow(Object.assign({
    icon: '/res/app.png',
    // smaller min size for "modal" windows
````

**Comment by Kilian** [comment]
I'm not sure if an absolute path also works (as opposed to including __dirname) but other than that, yeah that's how to get an icon on ubuntu :)

**Comment by luixxiul** [comment]
OK I'm pushing a PR for that

**Comment by luixxiul** [comment]
I don't know how to patch this, somebody please.

**Comment by luixxiul** [comment]
Stopped working. Restarting doesn't help.

**Comment by luixxiul** [comment]
On non packaged build it works.

**Comment by bbondy** [comment]
I think there's a cache @luixxiul so I think a reboot or a restart of the window manager needs to happen between testing.

I'm going to remove this from 0.10.4 but if there's a fix we can add it back in.
